.github/workflows/rebuild_dist.yml

```diff
@@ -0,0 +1,44 @@
+name: rebuild dist/ if needed
+
+on:
+  pull_request:
+
+jobs:
+  build:
+    runs-on: ubuntu-latest
+    steps:
+      - uses: actions/checkout@v3
+        with:
+          ref: ${{github.head_ref}}
+          persist-credentials: false # otherwise, the token used is the GITHUB_TOKEN, instead of your personal token
+          fetch-depth: 0 # otherwise, you will failed to push refs to dest repo
+      - run: npm install
```

Comment:
We can use `actions/cache` to cache our dependency :)
https://github.com/actions/cache

This could be helpful to reducing duration :)

Reply:
Yea. it makes sense, for now (in this PR) I will just add a setup-node with cache that should remove most of the heavy load from the operation. 

I agree putting the cache in place should be useful.

Reply:
<img width="1421" alt="image" src="https://user-images.githubusercontent.com/226559/204650828-50d9b331-b9c3-4bd1-bae3-42ebff88b3ab.png">
now you can see the cache being populated in Actions > cache